RefactoringsExperiments/src/main/java/ru/hse/kirilenko/refactorings/code_models/Fragment.java

```diff
@@ -0,0 +1,422 @@
+            features.add(new CSVItem(Feature.TotalLinesOfCode, getBodyLineLength()));
+        }
+
+        private void rankingScoreComputation(double c_length, double max_length) {
+            RankEvaluater ranker = new RankEvaluater(this, c_length, max_length);
+            this.score = ranker.getScore();
+        }
+
+        int analyzeDepth(String code) {
+            int dep = 0;
+            int area = 0;
+            int depInLine = 0;
+            for (Character ch : code.toCharArray()) {
+                if (ch == '{') {
+                    dep++;
+                    depInLine++;
+                } else if (ch == '}') {
+                    dep--;
+                    depInLine--;
+                } else if (ch == '\n') {
+                    int resDep = dep;
+                    if (depInLine > 0) {
+                        resDep--;
+                    }
+                    depInLine = 0;
+                    area += resDep;
+                }
+            }
+            return area;
+        }
+
+        private int getBodyLineLength() {
+            return (StringUtils.countMatches(getBody(), '\n') + 1);
+        }
+
+        private int getMethodLineLength() {
+            return (StringUtils.countMatches(getMethod(), '\n') + 1);
+        }
+
+        /**
+         * Returns body of the fragment (sequence of statements)
+         * w/o leading and trailing curly braces, whitespaces, linebreaks
+         */
+        private String getBody() {
+            return clearCode(this.methodDeclaration.getBody().toString());
+        }
+
+        private String getMethod() {
+            return this.methodDeclaration.toString();
+        }
+
+        private void writeFeatures(FileWriter fw) throws IOException {
+            int nFeatures = features.size();
+            features.sort(Comparator.comparingInt(ICSVItem::getId));
+            for (int i = 0; i < nFeatures; ++i) {
+                fw.append(String.valueOf(features.get(i).getValue()));
+                fw.append(';');
+            }
+            fw.append(String.valueOf(this.score)); //Score with 4 decimal places
+            fw.append(";");
+            fw.append("-\n");
+        }
+
+        private void process(FileWriter fw) throws IOException {
+            keywordFeaturesComputation();
+            historicalFeaturesComputation();
+            couplingFeaturesComputation();
+            methodDeclarationFeaturesComputation();
+            lengthFeaturesComputation();
+            rankingScoreComputation(0.1, 4);
+
+            writeFeatures(fw);
+        }
+
+    }
+
+    private boolean isValidStatement(Statement s) {
+        String linearText = clearCode(s.toString());
+        linearText = linearText.replace("\n", "").replace("\r", "");
+//        System.out.println(s.toString()+"\n-----------------------------------\n");
+        if (Pattern.matches(".*}", linearText) | Pattern.matches(".*;", linearText))
```

Comment:
simplify it to `return Pattern.matches(".*}", linearText) | Pattern.matches(".*;", linearText);`

Reply:
Here I thought that, although, your  suggestion is more tidy and objectively looks better, but is not as readable for the programmer. Nevertheless, will change it